src/nanoemoji/nanoemoji.py

```diff
@@ -24,497 +24,135 @@
-    try:
-        return SVG.parse(filename).topicosvg()
-    except Exception as e:
-        logging.warning(f"SVG.parse({filename}) failed: {e}")
-    return None
-
-
-def _inputs(filenames: Iterable[str]) -> Generator[InputGlyph, None, None]:
-    for filename in filenames:
-        codepoints = _codepoints_from_filename(os.path.basename(filename))
-        picosvg = _picosvg(filename)
-        if codepoints and picosvg:
-            yield InputGlyph(filename, codepoints, picosvg)
-
-
-def _ufo(family: str, upem: int, keep_glyph_names: bool = False) -> ufoLib2.Font:
-    ufo = ufoLib2.Font()
-    ufo.info.familyName = family
-    # set various font metadata; see the full list of fontinfo attributes at
-    # http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/
-    ufo.info.unitsPerEm = upem
-
-    # Must have .notdef and Win 10 Chrome likes a blank gid1 so make gid1 space
-    ufo.newGlyph(".notdef")
-    space = ufo.newGlyph(".space")
-    space.unicodes = [0x0020]
-    space.width = upem
-    ufo.glyphOrder = [".notdef", ".space"]
-
-    # use 'post' format 3.0 for TTFs, shaving a kew KBs of unneeded glyph names
-    ufo.lib[ufo2ft.constants.KEEP_GLYPH_NAMES] = keep_glyph_names
-
-    return ufo
-
-
-def _make_ttfont(config, ufo, color_glyphs):
-    if config.output_format == ".ufo":
-        return None
-
-    # Use skia-pathops to remove overlaps (i.e. simplify self-overlapping
-    # paths) because the default ("booleanOperations") does not support
-    # quadratic bezier curves (qcurve), which may appear
-    # when we pass through picosvg (e.g. arcs or stroked paths).
-    ttfont = None
-    if config.output_format == ".ttf":
-        ttfont = ufo2ft.compileTTF(ufo, overlapsBackend="pathops")
-    if config.output_format == ".otf":
-        cff_version = 1
-        if config.color_format.startswith("cff2_"):
-            cff_version = 2
-        ttfont = ufo2ft.compileOTF(
-            ufo, cffVersion=cff_version, overlapsBackend="pathops"
-        )
-
-    if not ttfont:
-        raise ValueError(
-            f"Unable to generate {config.color_format} {config.output_format}"
-        )
-
-    # Permit fixups where we can't express something adequately in UFO
-    _COLOR_FORMAT_GENERATORS[config.color_format].apply_ttfont(
-        ufo, color_glyphs, ttfont
-    )
+# internal flags, typically client wouldn't change
+flags.DEFINE_string("build_dir", "build/", "Where build runs.")
+flags.DEFINE_bool("gen_ninja", True, "Whether to regenerate build.ninja")
+flags.DEFINE_bool("exec_ninja", True, "Whether to run ninja.")
 
-    return ttfont
 
+def self_dir() -> str:
+    return os.path.dirname(os.path.abspath(__file__))
 
-def _write(ufo, ttfont, output_file):
-    logging.info("Writing %s", output_file)
 
-    if os.path.splitext(output_file)[1] == ".ufo":
-        ufo.save(output_file, overwrite=True)
-    else:
-        ttfont.save(output_file)
+def build_dir() -> str:
+    return os.path.abspath(FLAGS.build_dir)
```

Comment:
the default build_dir `./build` didn't exist in my repo and it caused a FileNotFoundError when attempting to read from the build.ninja file later.
I think this `build_dir()` function could call `os.makedirs(FLAGS.build_dir, exist_ok=True)` to ensure the requested build directory exists

Reply:
Fixed